protocol/clproto_cpp/src/clproto.cpp

```diff
@@ -1023,73 +1032,73 @@ template<> std::string encode<std::shared_ptr<State>>(const std::shared_ptr<Stat
       message = encode<State>(*obj);
       break;
     case StateType::SPATIAL_STATE:
-      message = encode<SpatialState>(*std::dynamic_pointer_cast<SpatialState>(obj));
+      message = encode<SpatialState>(*safe_dynamic_pointer_cast<SpatialState>(obj));
```

Comment:
I cannot think of a situation where this would have failed before (the ParameterInterface is the only problem I see), but I figured we can as well check here.